src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -650,6 +662,141 @@ vertexSet = Set.fromAscList . vertexList
 edgeSet :: (Ord a, Ord b) => AdjacencyMap a b -> Set.Set (a, b)
 edgeSet = Set.fromAscList . edgeList
 
+-- | The /circuit/ on a list of vertices.
+-- Complexity: /O(n * log(n))/ time and /O(n)/ memory.
+--
+-- @
+-- circuit []                       == 'empty'
+-- circuit [(x, y)]                 == 'edge' x y
+-- circuit [(x, y), (z, w)]         == 'connect' ('vertices' [x, z] []) ('vertices' [] [y, w])
```

Comment:
Should the right-hand side be `biclique [x, z] [y, w]`?

Reply:
Thanks, that looks better!